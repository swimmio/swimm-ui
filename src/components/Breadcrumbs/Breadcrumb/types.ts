export interface BreadcrumbType {
  name: string;
  icon: string;
  link: string | { link };
}
