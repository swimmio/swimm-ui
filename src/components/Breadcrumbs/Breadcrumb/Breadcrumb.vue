<template>
  <div class='breadcrumb'>
    <component :is='link ? "router-link" : "div"' :to='link' class='content' :class="{ 'has-link': !!link }">
      <Icon :name='icon' />
      <span>{{ name }}</span>
    </component>
  </div>
</template>

<script>
import Icon from '../../Icon/Icon';

export default {
  components: { Icon },
  props: {
    icon: { type: String, required: true },
    name: { type: String, required: true },
    link: {
      type: [ String, Object ]
    }
  }
};
</script>
<style scoped>
.breadcrumb {
  color: var(--text-color-primary);
  font-family: var(--fontfamily-main);
  font-size: var(--fontsize-xs);
  margin: 11px 0;
}

.content {
  display: inline-flex;
  align-items: center;
  color: var(--text-color-primary);
  padding: 4px;
  border-radius: 4px;
  transition: background-color 150ms linear;
}

.breadcrumb .content.has-link:hover {
  text-decoration: none;
  background-color: var(--color-hover);
}

.breadcrumb .content:not(.has-link) {
  cursor: default;
}

span {
  padding-inline-start: 6px;
}
</style>
