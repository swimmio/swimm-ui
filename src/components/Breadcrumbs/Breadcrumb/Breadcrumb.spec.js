import { shallowMount } from '@vue/test-utils';
import Breadcrumb from './Breadcrumb.vue';

describe('Breadcrumb', () => {
  it('should render item', async () => {
    const wrapper = shallowMount(Breadcrumb, {
      propsData: { icon: 'g', name: 'g' },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render router-link for link', async () => {
    const wrapper = shallowMount(Breadcrumb, {
      propsData: { icon: 'g', name: 'g', link: '/g' },
      slots: {
        default: '<router-link class="router-link"></router-link>',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
