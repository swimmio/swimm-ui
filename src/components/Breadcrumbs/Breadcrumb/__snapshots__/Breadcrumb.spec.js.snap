// Vitest Snapshot v1

exports[`Breadcrumb > should render item 1`] = `
"<div class=\\"breadcrumb\\" data-v-4117cb18=\\"\\">
  <div class=\\"content\\" data-v-4117cb18=\\"\\">
    <icon-stub name=\\"g\\" nopadding=\\"false\\" data-v-4117cb18=\\"\\"></icon-stub><span class=\\"name\\" data-v-4117cb18=\\"\\">g</span>
  </div>
</div>"
`;

exports[`Breadcrumb > should render router-link for link 1`] = `
"<div class=\\"breadcrumb\\" data-v-4117cb18=\\"\\">
  <router-link to=\\"/g\\" class=\\"content has-link\\" data-v-4117cb18=\\"\\">
    <icon-stub name=\\"g\\" nopadding=\\"false\\" data-v-4117cb18=\\"\\"></icon-stub><span class=\\"name\\" data-v-4117cb18=\\"\\">g</span>
  </router-link>
</div>"
`;
